src/rust/src/lazyframe/serde.rs

```diff
@@ -0,0 +1,12 @@
+use crate::{prelude::*, PlRLazyFrame, RPolarsErr};
+use savvy::{savvy, Result};
+
+#[savvy]
+fn deserialize_lf(json: &str) -> Result<PlRLazyFrame> {
+    let lp = serde_json::from_str::<DslPlan>(json).map_err(|_| {
+        let msg = "could not deserialize input into a LazyFrame";
+        RPolarsErr::Other(msg.to_string())
+    })?;
+    let out = LazyFrame::from(lp);
+    Ok(<PlRLazyFrame>::from(out))
+}
```

Comment:
This is not a clone of Python Polars' ones, is it?
The function should be the same as Python Polars.

Of course, it must be an associated function.

Reply:
No I think I had copied this from r-polars actually, so it should be modified/removed.

Deserialization is a bit tricky in terms of API and I think that's why we have deviated with this `deserialize_lf()` in r-polars. In py-polars they have:

```
pl.DataFrame.deserialize(...)
pl.LazyFrame.deserialize(...)
```

and I'm not sure neo-r-polars support something like this since `pl$DataFrame` just returns the body of the function. I wouldn't mind removing this serde file for now, what do you think?

Reply:
Since `pl$DataFrame` is a function, something like `pl$DataFrame$foo` cannot be realized.

However, this is a different thing: whether the function on the R side is something like `deserialize_lf`, an associated function can be used on the Rust side (extendr binds the associated functiona to the R side in a strange way, but with savvy, it is not a problem).

Reply:
I'm not sure I understand and I think this might be tricky to implement. I have removed this function for now: it's not essential and I don't want it to block the rest of the PR. I added a todo and we can address this later in a more focused PR.